x-pack/test/alerting_api_integration/spaces_only/tests/alerting/alerts_base.ts

```diff
@@ -135,6 +136,141 @@ instanceStateValue: true
       await taskManagerUtils.waitForActionTaskParamsToBeCleanedUp(testStart);
     });
 
+    it('should fire actions when an alert instance is resolved', async () => {
+      const reference = alertUtils.generateReference();
+
+      const { body: createdAction } = await supertestWithoutAuth
+        .post(`${getUrlPrefix(space.id)}/api/actions/action`)
+        .set('kbn-xsrf', 'foo')
+        .send({
+          name: 'MY action',
+          actionTypeId: 'test.noop',
+          config: {},
+          secrets: {},
+        })
+        .expect(200);
+
+      // pattern of when the alert should fire.
+      const pattern = {
+        instance: [false, true, true],
+      };
+
+      const createdAlert = await supertestWithoutAuth
+        .post(`${getUrlPrefix(space.id)}/api/alerts/alert`)
+        .set('kbn-xsrf', 'foo')
+        .send(
+          getTestAlertData({
+            alertTypeId: 'test.patternFiring',
+            schedule: { interval: '1s' },
+            throttle: null,
+            params: {
+              pattern,
+            },
+            actions: [
+              {
+                id: createdAction.id,
+                group: 'default',
+                params: {},
+              },
+              {
+                group: ResolvedActionGroup.id,
+                id: indexRecordActionId,
+                params: {
+                  index: ES_TEST_INDEX_NAME,
+                  reference,
+                  message: 'Resolved message',
+                },
+              },
+            ],
+          })
+        );
+
+      expect(createdAlert.status).to.eql(200);
+      const alertId = createdAlert.body.id;
+      objectRemover.add(space.id, alertId, 'alert', 'alerts');
+
+      const actionTestRecord = (
+        await esTestIndexTool.waitForDocs('action:test.index-record', reference)
+      )[0];
+
+      expect(actionTestRecord._source).to.eql({
+        config: {
+          unencrypted: `This value shouldn't get encrypted`,
+        },
+        secrets: {
+          encrypted: 'This value should be encrypted',
+        },
+        params: {
+          index: ES_TEST_INDEX_NAME,
+          reference,
+          message: 'Resolved message',
+        },
+        reference,
+        source: 'action:test.index-record',
+      });
```

Comment:
**optional nit:** we can remove the assertion of the content as it is done in the test above.

```suggestion
```

Reply:
+1